functions/uploads.py

```diff
@@ -201,36 +187,36 @@ def _gcs_add_prefix_reader_permission(group_email: str, prefix: str):
     )
 
     storage_client = storage.Client()
-    # Work-around for python gcloud client not able to get v3 policies:
     bucket = storage_client.get_bucket(GOOGLE_DATA_BUCKET)
-    policy_json = storage_client._connection.api_request(
-        method="GET",
-        path=f"/b/{GOOGLE_DATA_BUCKET}/iam",
-        query_params={"optionsRequestedPolicyVersion": 3},
-        _target_object=None,
-    )
 
-    # Work-around for python gcloud client not able to set v3 policies with conditions:
-    # We're adding them directly to the JSON request to API
+    # get v3 policy to use condition in bindings
+    policy = bucket.get_iam_policy(requested_policy_version=3)
+
+    # Set the policy's version to 3 to use condition in bindings.
+    policy.version = 3
+
     cleaned_prefix = prefix.replace('"', '\\"').lstrip("/")
-    # following https://cloud.google.com/iam/docs/reference/rest/v1/Policy
-    policy_json["bindings"].append(
+    grant_until_date = (
+        (datetime.now() + timedelta(GOOGLE_ANALYSIS_PERMISSIONS_GRANT_FOR_DAYS))
+        .date()
+        .isoformat()
+    )
+
+    # following https://github.com/GoogleCloudPlatform/python-docs-samples/pull/2730/files
+    policy.bindings.append(
         {
             "role": GOOGLE_ANALYSIS_GROUP_ROLE,
             "members": ["group:" + group_email],
             "condition": {
+                "title": f"Biofx analysis {prefix} on {datetime.now()}",
                 "description": "Auto-assigned from cidc-cloud-functions/uploads",
-                "expression": f'resource.name.startsWith("projects/_/buckets/{GOOGLE_DATA_BUCKET}/objects/{cleaned_prefix}")',
-                "title": f"Biofx analysis {prefix}",
+                "expression": f'resource.name.startsWith("projects/_/buckets/{GOOGLE_DATA_BUCKET}/objects/'
+                f'{cleaned_prefix}") && request.time < timestamp("{grant_until_date}T00:00:00")',
```

Comment:
Can you convert these two lines into variables with meaningful names? This is hard to read.
```python
prefixCheck = f'resource.name.startsWith("projects/_/buckets/{GOOGLE_DATA_BUCKET}/objects/{cleaned_prefix}")'
expiryCheck = f'request.time < timestamp("{grant_until_date}T00:00:00")'
...
"expression": f"{prefixCheck} && {expiryCheck}"
```